### PR TITLE
Integrate ML-IO library and apache arrow

### PR DESCRIPTION
**Pull request by rizwangilani**
*Description of changes:*
- Install miniconda3
- Update base and final Dockerfile.cpu files to integrate ML-IO library (high performance data access library for machine learning frameworks with support for multiple data formats)
- Update buildspec.yml
- Remove python version as an input argument
- Add apache arrow

*Testing:*
Built image successfully using CodeBuild and uploaded to ECR. Ran:
- unit tests
- integration tests using the new image in ECR
- manual notebook test that uses mlio for reading parquet

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/0.20.0/base/Dockerfile.cpu [hunk 1]
```diff
@@ -1,32 +1,25 @@
...
-RUN test $py_version || exit 1
-
 # Install python and other scikit-learn runtime dependencies
 # Dependency list from http://scikit-learn.org/stable/developers/advanced_installation.html#installing-build-dependencies
 RUN apt-get update && \
-    apt-get -y install build-essential libatlas-dev git wget curl nginx jq && \
-    if [ $py_version -eq 2 ]; \
-       then apt-get -y install python-dev python-setuptools \
-                     python-numpy python-scipy libatlas3-base; \
-       else apt-get -y install python3-dev python3-setuptools \
```
Comment (ericangelokim) [inline docker/0.20.0/base/Dockerfile.cpu]: Don't you need to apt-get these?

File: docker/0.20.0/base/Dockerfile.cpu [hunk 2]
```diff
@@ -1,32 +1,25 @@
...
+    rm Miniconda3-latest-Linux-x86_64.sh
+
+ENV PATH=/miniconda3/bin:${PATH}
 
-# Install pip
-RUN cd /tmp && \
-     curl -O https://bootstrap.pypa.io/get-pip.py && \
-     if [ $py_version -eq 2 ]; \
-        then python2 get-pip.py; \
-        else python3 get-pip.py; fi && \
```
Comment (ericangelokim) [inline docker/0.20.0/base/Dockerfile.cpu]: Same for this comment.

**Comment by ericangelokim** [comment]
Also, please describe tests run.

File: docker/0.20.0/base/Dockerfile.cpu (hunk 2 above)
Comment (rizwangilani) [reply docker/0.20.0/base/Dockerfile.cpu]: installed under miniconda.

File: docker/0.20.0/base/Dockerfile.cpu [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
-     rm get-pip.py
+RUN conda update -y conda && \
+    conda install protobuf tbb libiconv && \
```
Comment (cbalioglu) [inline docker/0.20.0/base/Dockerfile.cpu]: You don't need this line. `mlio-py` installs them as its dependencies.

File: docker/0.20.0/base/Dockerfile.cpu [hunk 4]
```diff
@@ -1,32 +1,25 @@
...
+    conda install -c conda-forge pyarrow=0.14.1 && \
+    conda install -c mlio -c conda-forge mlio-py && \
+    pip install scipy
```
Comment (cbalioglu) [inline docker/0.20.0/base/Dockerfile.cpu]: scipy can also be installed via conda.

Comment (rizwangilani) [reply docker/0.20.0/base/Dockerfile.cpu]: done

File: docker/0.20.0/base/Dockerfile.cpu (hunk 3 above)
Comment (rizwangilani) [reply docker/0.20.0/base/Dockerfile.cpu]: done

**Pull request merged**